### PR TITLE
Share screen with state

### PR DESCRIPTION
**Pull request by ibeckermayer**
Added an `InfoBanner` to be displayed when a user comes in contact with somebody reporting COVID. 

Biggest change: I renamed `Share` to `Home` and added a rudimentary struct
```swift
struct UserState {
    var firstTimeUser: Bool
    var hasBeenInContact: Bool
    var hasBeenTestedInLast14Days: Bool
}
```
This struct should eventually be represented by the global stage manager, but for now it's here so that I can work on the state logic for the Home screen. The logic is a bit complex, though I've tried to keep it as straightforward as I can.

I also spun a function in `Button` to make it simpler to place buttons centered between two vertical pixels 
```swift
func drawBetween(parentVC: UIViewController, 
                 top: CGFloat, 
                 bottom: CGFloat, 
                 centerX: CGFloat)
```
I figure that even if we end up rewriting everything with storyboards and constraints, this function will come in handy because it accounts for the optional button subtext, which I imagine is tricky to do from the storyboards.

### CODE REVIEW

**Comment by ibeckermayer** [comment]
@jeffctown Any suggestions for what I should about these long string literals and the (necessarily) long function that are breaking our linter rules?

**Comment by jeffctown** [comment]
@ibeckermayer to get rid of the long string SwiftLint warnings you could put them in a storyboard (preferred), try a one line [multiline literal](https://www.swiftbysundell.com/tips/multiline-string-literals/), or just [disable the warnings](https://github.com/covid19risk/covidwatch-ios/blob/619b852d3e0954e9287728092d6c141e9755c667/COVIDWatch%20iOS/ViewControllers/Test.swift#L93).

File: COVIDWatch iOS/UIComponents/Button.swift [hunk 1]
```diff
@@ -74,6 +74,18 @@ class Button: UIView {
...
 
+    func drawBetween(parentVC: UIViewController, top: CGFloat, bottom: CGFloat, centerX: CGFloat) {
+//        Draw the button so that it and its subtext taken together are
+//        centered between top and bottom
+        self.draw(parentVC: parentVC, centerX: centerX, centerY: (top+bottom)/2)
+        if self.subtext != nil {
+            let adjustment = (self.subtext!.frame.maxY - self.frame.maxY) / 2
+            self.center.y -= adjustment
+            self.text.center.y -= adjustment
+            self.subtext!.center.y -= adjustment
```
Comment (jeffctown) [inline COVIDWatch iOS/UIComponents/Button.swift]: Can these be if lets instead of force unwrapping please?

**Comment by therealnikhil** [comment]
This is behind dev, could you merge and submit a new PR? The reason I ask this is because I used UIColor extensions to update the color of buttons to v1.6 design. The use of UIColor fixed colors rather than hardcoding hex values helps with future design updates as well.

File: COVIDWatch iOS/UIComponents/Button.swift (hunk 1 above)
Comment (ibeckermayer) [reply COVIDWatch iOS/UIComponents/Button.swift]: I don't understand what that suggestion means (Swift noob)

Comment (ibeckermayer) [reply COVIDWatch iOS/UIComponents/Button.swift]: Never mind I read it wrong

**Review by therealnikhil — APPROVED** [review]
Looks good!

**Pull request merged**